From 75b24bae6bcb7e33ee9b564d751e0523602ad040 Mon Sep 17 00:00:00 2001
From: Josua Mayer <josua@solid-run.com>
Date: Sat, 23 Dec 2023 21:30:24 +0100
Subject: [PATCH 4/5] arm: mvebu: clearfog: upgrade sata speed to sata-3
 (6Gbps)

According to Marvell documentation Amrada 385 and 388 support up to
sata-3. Upgrade the serdes rates accordingly.

Signed-off-by: Josua Mayer <josua@solid-run.com>
---
 board/solidrun/clearfog/clearfog.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/board/solidrun/clearfog/clearfog.c b/board/solidrun/clearfog/clearfog.c
index 6fa2fe5fe3..2ee98a3020 100644
--- a/board/solidrun/clearfog/clearfog.c
+++ b/board/solidrun/clearfog/clearfog.c
@@ -51,7 +51,7 @@ static void cf_read_tlv_data(void)
 
 /* The starting board_serdes_map reflects original Clearfog Pro usage */
 static struct serdes_map board_serdes_map[] = {
-	{SATA0, SERDES_SPEED_3_GBPS, SERDES_DEFAULT_MODE, 0, 0},
+	{SATA0, SERDES_SPEED_6_GBPS, SERDES_DEFAULT_MODE, 0, 0},
 	{SGMII1, SERDES_SPEED_1_25_GBPS, SERDES_DEFAULT_MODE, 0, 0},
 	{PEX1, SERDES_SPEED_5_GBPS, PEX_ROOT_COMPLEX_X1, 0, 0},
 	{USB3_HOST1, SERDES_SPEED_5_GBPS, SERDES_DEFAULT_MODE, 0, 0},
@@ -76,14 +76,14 @@ int hws_board_topology_load(struct serdes_map **serdes_map_array, u8 *count)
 
 	if (IS_ENABLED(CONFIG_CLEARFOG_CON2_SATA)) {
 		board_serdes_map[4].serdes_type = SATA2;
-		board_serdes_map[4].serdes_speed = SERDES_SPEED_3_GBPS;
+		board_serdes_map[4].serdes_speed = SERDES_SPEED_6_GBPS;
 		board_serdes_map[4].serdes_mode = SERDES_DEFAULT_MODE;
 		board_serdes_map[4].swap_rx = 1;
 	}
 
 	if (IS_ENABLED(CONFIG_CLEARFOG_CON3_SATA)) {
 		board_serdes_map[2].serdes_type = SATA1;
-		board_serdes_map[2].serdes_speed = SERDES_SPEED_3_GBPS;
+		board_serdes_map[2].serdes_speed = SERDES_SPEED_6_GBPS;
 		board_serdes_map[2].serdes_mode = SERDES_DEFAULT_MODE;
 		board_serdes_map[2].swap_rx = 1;
 	}
-- 
2.35.3

